pathology/hovernet/training_ignite.py

```diff
@@ -0,0 +1,404 @@
+    train_data = [data[i] for i in splits[fold]['train']]
+    valid_data = [data[i] for i in splits[fold]['valid']]
+
+    return train_data, valid_data
+
+def _from_engine(keys):
+    keys = ensure_tuple(keys)
+
+    def _wrapper(data):
+        ret = [[i[k][HoVerNetBranch.NP.value] for i in data] for k in keys]
+        return tuple(ret) if len(ret) > 1 else ret[0]
+
+    return _wrapper
+
+def post_process(output, return_binary=True, return_centroids=False, output_classes=None):
+    pred = output["pred"]
+    device = pred[HoVerNetBranch.NP.value].device
+    if HoVerNetBranch.NC.value in pred.keys():
+        type_pred = Activations(softmax=True)(pred[HoVerNetBranch.NC.value])
+        type_pred = AsDiscrete(argmax=True)(type_pred)
+
+    post_trans_seg = Compose([
+        GenerateWatershedMaskd(keys=HoVerNetBranch.NP.value, softmax=True),
+        GenerateInstanceBorderd(keys='mask', hover_map_key=HoVerNetBranch.HV, kernel_size=3),
+        GenerateDistanceMapd(keys='mask', border_key='border', smooth_fn=GaussianSmooth()),
+        GenerateWatershedMarkersd(keys='mask', border_key='border', threshold=0.7, radius=2, postprocess_fn=FillHoles()),
+        Watershedd(keys='dist', mask_key='mask', markers_key='markers')
+    ])
+    pred_inst_dict = post_trans_seg(pred)
+    pred_inst = pred_inst_dict['dist']
+
+    inst_id_list = np.unique(pred_inst)[1:]  # exclude background
+
+    inst_info_dict = None
+    if return_centroids:
+        inst_info_dict = {}
+        for inst_id in inst_id_list:
+            inst_map = pred_inst == inst_id
+            inst_bbox = BoundingRect()(inst_map)
+            inst_map = inst_map[:, inst_bbox[0][0]: inst_bbox[0][1], inst_bbox[0][2]: inst_bbox[0][3]]
+            offset = [inst_bbox[0][2], inst_bbox[0][0]]
+            inst_contour = GenerateInstanceContour()(inst_map, offset)
+            inst_centroid = GenerateInstanceCentroid()(inst_map, offset)
+            if inst_contour is not None:
+                inst_info_dict[inst_id] = {  # inst_id should start at 1
+                    "bounding_box": inst_bbox,
+                    "centroid": inst_centroid,
+                    "contour": inst_contour,
+                    "type_probability": None,
+                    "type": None,
+                }
+
+    if output_classes is not None:
+        for inst_id in list(inst_info_dict.keys()):
+            inst_type, type_prob = GenerateInstanceType()(
+                bbox=inst_info_dict[inst_id]["bounding_box"], type_pred=type_pred, seg_pred=pred_inst, instance_id=inst_id)
+            inst_info_dict[inst_id]["type"] = inst_type
+            inst_info_dict[inst_id]["type_probability"] = type_prob
+
+    pred_inst = convert_to_tensor(pred_inst, device=device)
+    if return_binary:
+        pred_inst[pred_inst > 0] = 1
+    output["pred"][HoVerNetBranch.NP.value] = pred_inst
+    output["pred"]["inst_info_dict"] = inst_info_dict
+    output["pred"]["pred_inst_dict"] = pred_inst_dict
+    return output
```

Comment:
@KumoLiu, I think this post processing should become a component in `apps/pathology/transforms`. Otherwise, I don't know how we can implement this logic in a monai bundle, do you?

Please check `train.json` from any of the models [here](https://github.com/Project-MONAI/model-zoo/tree/dev/models) and see how a bundle is structured. It is mostly the name of components to use and not code logic.

@Nic-Ma, do you have other opinion here?

Reply:
@KumoLiu here if we are not providing the user the real flexibility to modify the post-processing, I believe it would be better to have one single components in `apps/pathology/transforms` as `HoVerNetPostProcessing`. What do you think?

Reply:
Hi @drbeh, I just create a [PR #5462](https://github.com/Project-MONAI/MONAI/pull/5462) to combine all the postprocessing components, for any suggestions feel free to take it out, thanks!